packages/react-test-renderer/src/ReactTestRendererScheduling.js

```diff
@@ -80,7 +80,33 @@ export function flushThrough(expectedValues: Array<mixed>): Array<mixed> {
   }
   if (yieldedValues === null) {
     // Always return an array.
-    return [];
+    yieldedValues = [];
+  }
+  if (yieldedValues.length !== expectedValues.length) {
```

Comment:
Isn't the loop below sufficient?

Reply:
Yeah. I thought this error message might be more readable, but you're right- it's unnecessary.